File: python/challenges/slugify.py
import re


def slugify(s: str) -> str:
    return re.sub(r"['\"!@#$%^&\*\(\)\[\]\{\};:\,\./<>\?\|`~=_+ ]", "-", s.lower()).strip("-")


def deslugify(s: str) -> str:
    return " ".join([part.capitalize() for part in s.split("-")])

File: elixir/apps/challenges/lib/challenges/slugify.ex
defmodule Challenges.Slugify do
  @spec slugify(String.t()) :: String.t()
  def slugify(str) do
    str
    |> String.downcase()
    |> String.replace(~r/['"!@#$%^&\*\(\)\[\]\{\};:\,\.\/<>\?\|`~=_+ ]/, "-")
    |> String.trim("-")
  end

  @spec deslugify(String.t()) :: String.t()
  def deslugify(str) do
    String.split(str, ~r/\-/)
    |> Enum.map(fn s -> String.capitalize(s) end)
    |> Enum.join(" ")
  end
end
